circuits/cpp/barretenberg/py/src/estimation/circuit.py

```diff
@@ -0,0 +1,73 @@
+from flavors import *
+from utils import info, DataLog
+from msm import MSM
+
+
+class CircuitNoPCS:
+    """
+    @brief A circuit whose recursive verification cost is of interest to us.
+    The class will later be completed with a PCS through inheritance.
+
+    @param flavor: The flavor of Honk in which we're doing recursion.
+    @param log_n: The log of the smallest power of 2 greater than the circuit size.
+    @num_pub_inputs: The number of public inputs to the circuit.
+    """
+
+    def __init__(self, flavor, log_n, num_pub_inputs):
+        self.flavor = flavor
+        self.log_n = log_n
+        self.num_public_inputs = num_pub_inputs
+
+        # Each polynomial consists of n 32-byte field elements. After the first
+        # round of sumcheck, we also have a partial evaluation oach polynomials,
+        # which consists of n/2 32-byte field elements
+        self.max_memory = int(flavor.NUM_POLYNOMIALS *
+                              (1 << log_n) * 32 * 1.5)  # bytes
```

Comment:
Dunno if this is too much detail/context for this tool, but one project on my mind is reducing the memory of the sumcheck protocol at the expense of some more computations. 

For example, instead of folding the input polynomials into size 1/2 polynomials after the 1st round, we can derive the 2nd round values from the 1st round polynomials, and then fold (using two challenges) into size 1/4 polynomials.

Might be helpful if RAM issues become a concern

Also a lot of polynomials (about half) in the ECCVM are boolean, I don't know if that is something worth considering when estimating max memory